### PR TITLE
Remove deprecated admin/ajax-tab.php

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Removed deprecated admin-dev/ajax.php file
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes - the file has been deleted
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | Quite a lot of requests have changed in obscure parts of the software that I'm unable to track, but they all changed the same per convention, so we can guess that if it works for 1, it works for N, right?
| Possible impacts? | This changes how some Ajax requests are made, so they either work, or they don't. Read below 

## Breaking changes

- The file `admin-dev/ajax-tab.php` has been deleted.

All calls to that file have been changed to `index.php?ajax=1`

## Proof it works for 1!

Go to Customer Settings > Groups, edit one, then click on "add category discount". If the tree loads, then that's one request I didn't mess up!

Screenshot 2021-10-18 at 17 16 49

## Other impacted areas in BO that I didn't test but should be working

- Upload a logo in the carrier wizard
- Load stuff in the cart rules management part
- Mark customer thread as read
- Forgot password form in login can be sent
- Product category tree loads in... somewhere

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26294)

### CODE REVIEW

File: js/admin/login.js [hunk 1]
```diff
@@ -226,7 +226,7 @@ function doAjaxForgot() {
 		$.ajax({
 			type: 'POST',
 			headers: {'cache-control': 'no-cache'},
-			url: 'ajax-tab.php' + '?rand=' + new Date().getTime(),
+			url: 'index.php' + '?rand=' + new Date().getTime(),
```
Comment (Progi1984) [inline js/admin/login.js]:
```suggestion
			url: 'index.php?rand=' + new Date().getTime(),
```

File: js/admin/login.js [hunk 2]
```diff
@@ -257,7 +257,7 @@ function doAjaxReset() {
...
```
Comment (Progi1984) [inline js/admin/login.js]:
```suggestion
			url: 'index.php?rand=' + new Date().getTime(),
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback, my boss

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by atomiix** [comment]
@eternoendless can you rebase your PR, please? :)

**Comment by eternoendless** [comment]
Rebased

**Review by matthieu-rolland — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @eternoendless,

Could you please check the Conflicting files?

Thanks!

**Comment by eternoendless** [comment]
Rebased

**Review by matks — APPROVED** [review]
Change are OK for me.

ping @Quetzacoalt91 is this something that might be an issue for modules ?

**Comment by Quetzacoalt91** [comment]
> ping @Quetzacoalt91 is this something that might be an issue for modules ?

I don't have any example in mind but as this has been deprecated and you're going to remove it for the next major, I guess we're fine if the documentation remind the community about this deletion.

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Could you document your BC Break, please ?

After that, it will be an approval for me.

**Comment by eternoendless** [comment]
Done in description @Progi1984

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @eternoendless,

It is ok :heavy_check_mark: 
https://watch.screencastify.com/v/fW2K3ImV4mBfrEoIwgLq
https://watch.screencastify.com/v/5ZgBlZm6jBsBNy24IFtK

I tried to make different cases.

- Go to Customer Settings > Groups, edit one, then click on "add category discount".  => ok
- Upload a logo in the carrier wizard > ok (edit a carrier & add new carrier)
- Load stuff in the cart rules management part (create a cart rule from BO & from Order > Add new order)
- Mark customer thread as read (make a forward to another employee)
- Forgot password form in login can be sent (tested from BO & FO) 
- Product category tree loads in... somewhere (I checked FO > category page > category_tree module / BO > Catalog > Products > select & unselect a category)


Thanks!

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @eternoendless & @khouloudbelguith

**Pull request merged**